app/uk/gov/hmrc/helptosavefrontend/connectors/EmailVerificationConnector.scala

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavefrontend.connectors
+
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import org.joda.time.Period
+import play.api.Configuration
+import play.api.http.Status._
+import play.api.libs.json.{JsValue, Json}
+import uk.gov.hmrc.helptosavefrontend.models.VerifyEmailStatus._
+import uk.gov.hmrc.helptosavefrontend.models.{EmailVerificationRequest, VerifyEmailStatus}
+import uk.gov.hmrc.helptosavefrontend.util.Logging
+import uk.gov.hmrc.play.config.ServicesConfig
+import uk.gov.hmrc.play.http.{HeaderCarrier, HttpResponse}
+import uk.gov.hmrc.play.http.ws.WSHttp
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+@ImplementedBy(classOf[EmailVerificationConnectorImpl])
+trait EmailVerificationConnector {
+
+  def verifyEmail(nino: String, newEmail: String)(implicit hc: HeaderCarrier): Future[VerifyEmailStatus]
+  def isVerified(email: String)(implicit hc: HeaderCarrier): Future[Either[VerifyEmailStatus, Boolean]]
+}
+
+@Singleton
+class EmailVerificationConnectorImpl @Inject() (http: WSHttp, conf: Configuration) extends EmailVerificationConnector with ServicesConfig with Logging {
+
+  val emailVerifyBaseURL = baseUrl("email-verification")
+  val verifyEmailURL = s"$emailVerifyBaseURL/email-verification/verification-requests"
+  def isVerifiedURL(email: String) = s"$emailVerifyBaseURL/email-verification/verified-email-addresses/$email"
+
+  val continueURL = "email-verification.continue.baseUrl"
+  //+ EmailVerificationController.showSuccess.
+  val defaultTemplate = "hts_email_verification"
+
+  override def verifyEmail(nino: String, newEmail: String)(implicit hc: HeaderCarrier): Future[VerifyEmailStatus] = {
+    conf.getInt("services.email-verification.linkTTLMinutes") match {
```

Comment:
also, instead of using JODA time and reading minutes you can do this in the config:
```
x = 30 minutes
```
then you can use the configs library to do
```
  import configs.syntax._
  val ttl: FiniteDuration = conf.underlying.get[FiniteDuration]("x").valueOrElse(sys.error("Could not read x"))
```